src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/Process_StreamTests.cs

```diff
@@ -42,17 +43,34 @@ public void Process_SyncErrorStream()
         [Fact, ActiveIssue(1538, PlatformID.OSX)]
         public void Process_AsyncErrorStream()
         {
-            StringBuilder sb = new StringBuilder();
-            Process p = CreateProcessError();
-            p.StartInfo.RedirectStandardError = true;
-            p.ErrorDataReceived += (s, e) => { sb.Append(e.Data); };
-            p.Start();
-            p.BeginErrorReadLine();
+            {
+                StringBuilder sb = new StringBuilder();
+                Process p = CreateProcessError();
+                p.StartInfo.RedirectStandardError = true;
+                p.ErrorDataReceived += (s, e) => { sb.Append(e.Data); };
+                p.Start();
+                p.BeginErrorReadLine();
+
+                if (p.WaitForExit(WaitInMS))
+                    p.WaitForExit(); // This ensures async event handlers are finished processing.
+
+                Assert.Equal(TestExeName + " error stream", sb.ToString());
+            }
 
-            if (p.WaitForExit(WaitInMS))
-                p.WaitForExit(); // This ensures async event handlers are finished processing.
+            {
+                // Now add the CancelAsyncAPI as well.
+                StringBuilder sb = new StringBuilder();
+                Process p = CreateProcessError();
+                p.StartInfo.RedirectStandardError = true;
+                p.ErrorDataReceived += (s, e) => { sb.Append(e.Data); ((Process)s).CancelErrorRead(); };
```

Comment:
What's being tested here?

And if the only difference between the two versions is whether CancelErrorRead is called, did you consider just putting the whole test in a loop for the two cases, executing the CancelErrorRead call on just one iteration?